runs/atreboot.sh

```diff
@@ -84,15 +84,23 @@ if (( rseenabled == 1 )); then
 fi
 
 # check if rfid is configured and start daemons to listen on input devices
+
+# Kill only if process exists, avoid error messages
+if ps ax |grep -v grep |grep "python /var/www/html/openWB/runs/readrfid.py" > /dev/null
+then
+	sudo kill $(ps aux |grep '[i]readrfid.py' | awk '{print $2}')
+fi
+if ps ax |grep -v grep |grep "python /var/www/html/openWB/runs/readrfid2.py" > /dev/null
+then
+	sudo kill $(ps aux |grep '[i]readrfid2.py' | awk '{print $2}')
+fi
```

Comment:
Es ginge auch einfacher:
```bash
pkill -f 'python /var/www/html/openWB/runs/readrfid2?\.py'
```

Reply:
Ja hast recht.
Das "r" gehört in die klammer (kopierfehler meinerseits)
also [r]eadrfid

Optimieren kann man vieles. 
Hier das ist eher Hausmanskost kopiert von den anderen Stellen im Script.

Hm..... Wie kann ich nun den PR ändern?


PS:
Ab wann gibt's pkill ?

Reply:
> Optimieren kann man vieles. Hier das ist eher Hausmanskost kopiert von den anderen Stellen im Script.

Man sieht dem OpenWB-Code leider an jeder Ecke an, dass er zu großen Teilen per copy&paste entstanden ist. Robuster und wartbarer macht das den Code nicht.

Wo wir uns schonmal über Performance unterhalten haben: Performanter auch nicht. Der Code oben ruft 2x`ps`, 3x`grep`, 1x`awk`, 1x`sudo`, 1x`kill` = 8 Prozesse auf um seine Arbeit zu verrichten - pro Script, welches gekillt werden soll. Also insgesamt 16 Programmstarts. Ist zwar alles kein Drama... Aber `pkill` löst das ganze in einem einzigen kurzen und verständlichen Aufruf.

> Hm..... Wie kann ich nun den PR ändern?

Einfach wieder auf den Branch pushen. Ggf. commit neu machen (`git reset --hard master`, dann neu commiten) und force-pushen `git push -f origin`.

Nebenbei wäre eine etwas aussagekräftigere Commitmessage als "Update atreboot.sh" wünschenswert. Leute gucken sich den Code in Monaten oder Jahren noch an und fragen sich warum etwas so gemacht wurde wie es ist. `git blame` hilft ungemein, wenn dann die commit-message keinen Hinweis auf die Motivation liefert lässt das großes Potential ungenutzt verstreichen.

> 
> PS: Ab wann gibt's pkill ?

Was meinst du mit "ab wann"? `pkill` kommt genau wie `kill` aus dem Paket "procps". Ob das vor 20 Jahren schon so war weiß ich nicht.

Reply:
Nun ja im gegensatz zu regel.sh und seinen modulen wird das ja nur einmal je Reboot durchgeführt.
Von daher sehe ich das timing hier entspanter.
Das Kommando kill ist in BASH eingebaut. pkill soweit ich weiss nur ein externer Befehl.